File: 백준/Silver/5567. 결혼식/결혼식.cc
#include <bits/stdc++.h>
using namespace std;

vector<int> adj[504];
int n, m, visited[504], u, v, num;

int main() {
    cin >> n;
    cin >> m;
    while(m--) {
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }

    queue<int> q;
    q.push(1);
    visited[1] = 1;
    while(q.size()) {
        int cur = q.front();
        q.pop();
        for(auto nxt : adj[cur]) {
            if(visited[nxt])continue;
            visited[nxt] = visited[cur] + 1;
            q.push(nxt);
        }
    }
    for(int i = 1; i <= n; i++) {
        if(visited[i] && visited[i] <= 3) num++;
    }

    cout << num - 1;
}

File: 백준/Silver/5567. 결혼식/결혼식.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;
import java.util.stream.IntStream;

public class Main {
    static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    static int[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());

        visited = new int[n + 1];
        IntStream.rangeClosed(0, n).forEach(i -> visited[i] = -1);

        for (int i = 0; i < n + 1; i++) {
            graph.add(new ArrayList<>());
        }

        for (int i = 0; i < m; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int u = Integer.parseInt(st.nextToken());
            int v = Integer.parseInt(st.nextToken());

            graph.get(u).add(v);
            graph.get(v).add(u);
        }

        Queue<Integer> queue = new LinkedList<>();
        queue.add(1);
        visited[1] = 0;
        while (!queue.isEmpty()) {
            int cur = queue.poll();
            for (int nxt : graph.get(cur)) {
                if (visited[nxt] != -1) continue;
                queue.add(nxt);
                visited[nxt] = visited[cur] + 1;
            }
        }
        int count = 0;
        for (int i = 1; i <= n; i++) {
            if (visited[i] == 1 || visited[i] == 2) {
                count++;
            }
        }

        System.out.println(count);



    }
}